lib/dpram.js

```diff
@@ -37,15 +43,23 @@ InModuleBody {
   val ${e.name}0 = ${e.name}Node.bundle
   val ${e.name}Mem = SyncReadMem(${addrSpace}, Vec(${banks}, UInt((${dataBankWidth}).W)))
 
-  val address = ${io.ADDR}
+  val writeAddress = ${io.WRADDR}
   val writeData = VecInit(${io.WRDATA}.toBools.grouped(${dataBankWidth}).map(VecInit(_).asUInt).toSeq)
   val byteEnable = ${get(portMaps, ['BEN'], null) ? `${io.BEN}.toBools` : 'VecInit(Seq(true.B))'}
 
-  ${io.RDDATA} := ${e.name}Mem.read(${io.RDADDR}, ${io.RDEN}).asUInt
+${portMaps.RDCLK ? `  val readClock = ${io.RDCLK}.asClock()
```

Comment:
Maybe put the `val readClock` on its own line, like what you did with `spram.js`? It's a bit difficult to read this because of all the nested interpolations, multiline strings, and careful indentation handling.